aquacomputer_d5next.c

```diff
@@ -539,6 +569,9 @@ static int aqc_get_ctrl_val(struct aqc_data *priv, int offset, long *val, size_t
 		goto unlock_and_return;
 
 	switch (size) {
+	case 32:
```

Comment:
Add a paramater `bool is_le` to the function and propagate the cases for all sizes. Same for the buffer setting function.

Reply:
I am thinking about using an enum or constants which specify size and endianness, as we currently only need three different options.

Reply:
But we only need three options, so I don't see how an enum would help, it would only overcomplicate stuff. Adding a boolean is just a few lines more in the switch case for size 16. Similarly to the other thing, it was Guenter who requested that the size be a direct parameter (in some previous patches where it was more convoluted).

Reply:
OK, good point. In the end my biggest concern is Guenter accepting the code.